packages/strapi-generate-new/lib/after.js

```diff
@@ -46,52 +50,71 @@ module.exports = (scope, cb) => {
   const dependencies = _.get(packageJSON, 'dependencies');
   const strapiDependencies = Object.keys(dependencies).filter(key => key.indexOf('strapi') !== -1);
   const othersDependencies = Object.keys(dependencies).filter(key => key.indexOf('strapi') === -1);
-  const globalRootPath = execSync('npm root -g');
+  // Add this check to know if we are in development mode so the creation is faster.
+  const isStrapiInstalledWithNPM = packageManager.isStrapiInstalledWithNPM();
+  const globalRootPath = isStrapiInstalledWithNPM ? execSync('npm root -g') : execSync(packageManager.commands('root -g'));
```

Comment:
```js
const globalRootPath = packageManager.commands('root -g');
````

In the `commands` function, it already verifies is the package has been installed using npm or not?